tests/markov/msm/test_tram.py

```diff
@@ -345,14 +372,64 @@ def test_unknown_connectivity():
     ]
 )
 def test_invalid_input(dtrajs, bias_matrices, ttrajs):
-    dtrajs = [np.asarray(traj) for traj in dtrajs]
+    dtrajs, bias_matrices, ttrajs = to_numpy_arrays(dtrajs, bias_matrices, ttrajs)
+    tram = TRAM()
 
-    if ttrajs is not None:
-        ttrajs = [np.asarray(traj) for traj in ttrajs]
+    with np.testing.assert_raises(ValueError):
+        tram._validate_input(ttrajs, dtrajs, bias_matrices)
 
-    if not isinstance(bias_matrices, np.ndarray):
-        bias_matrices = [np.asarray(M) for M in bias_matrices]
 
+@pytest.mark.parametrize(
+    "dtrajs, bias_matrices, ttrajs",
+    [
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((2, 3, 3)), None),
+        ([[0, 0, 0], [0, 0]], np.zeros((2, 3, 2)), None),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((2, 2, 2)), None),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((3, 2, 2)), None),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((1, 2, 2)), None),
+        ([[0, 0, 0], [0, 0]], [[[0, 0], [0, 0], [0, 0]], [[0, 0], [0, 0, 0]]], None),
+        ([[0, 0, 0], [0, 'x', 0]], np.zeros((2, 3, 3)), None),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((2, 3, 2)), [[0, 0, 0], [0, 1, 2]]),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((2, 3, 2)), [[0, 0], [1, 1, 1]]),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((2, 3, 2)), [[0, 0, 'x'], [1, 1, 1]]),
+        ([[0, 0, 0], [0, 0, 0]], np.zeros((2, 3, 2)), [[0, 0, 0], [0, 1]]),
+        ([[0, 1, 0], [0, 0, 0]], np.zeros((2, 3, 2)), [[0, 0, 0], [0, 0, 0]]),
+        ([[0, 0, 0], [0, 1, 0]], np.zeros((2, 3, 2)), [[0, 2, 0], [0, 0, 0]]),
+    ]
+)
+def test_invalid_input_initialized_from_model(dtrajs, bias_matrices, ttrajs):
     tram = TRAM()
+    tram.n_therm_states = 2
+    tram.n_markov_states = 1
+
+    dtrajs, bias_matrices, ttrajs = to_numpy_arrays(dtrajs, bias_matrices, ttrajs)
+
     with np.testing.assert_raises(ValueError):
         tram._validate_input(ttrajs, dtrajs, bias_matrices)
+
+@pytest.mark.parametrize(
+    "dtrajs, bias_matrices, ttrajs",
+    [
+        ([[0, 0, 1], [0, 1, 0]], np.zeros((2, 3, 2)), [[0, 0, 0], [0, 0, 0]])
+    ]
+)
+def test_valid_input_initialized_from_model(dtrajs, bias_matrices, ttrajs):
+    tram = TRAM()
+    tram.n_therm_states = 2
+    tram.n_markov_states = 2
+
+    dtrajs, bias_matrices, ttrajs = to_numpy_arrays(dtrajs, bias_matrices, ttrajs)
+
+    tram._validate_input(ttrajs, dtrajs, bias_matrices)
```

Comment:
if `_validate_input` is private then it's an implementation detail and shouldn't be tested directly